### ISSUE TITLE
Add benchmark as cm_generate_makefile option

### ISSUE THREAD

**Issue by brian-kelley** [kokkos/kokkos-kernels#1761]
@cz4rs @lucbv

It would be nice if cm_generate_makefile had a flag like ``--with-benchmark`` to set ``-DKokkosKernels_ENABLE_BENCHMARK=ON``.

**Comment by lucbv** [comment]
Yeah, I have been purposely refraining from doing it just yet as we are still making some heavy changes and I do not want folks to mistakenly believe that because the option is in place it is fully ready...
I believe we are getting pretty close to that point though.

**Comment by ndellingwood** [comment]
Completed in https://github.com/kokkos/kokkos-kernels/pull/2945
